api/app/signals/apps/reporting/csv/datawarehouse/tasks.py

```diff
@@ -54,3 +54,42 @@ def save_csv_files_datawarehouse():
         save_csv_file_datawarehouse(create_kto_feedback_csv)
     except EnvironmentError:
         pass
+
+
+@app.task
+def save_csv_file_endpoint(func: Callable[[str], str]) -> None:
+    """
+    Create CSV files for csv endpoint and save them on the storage backend.
+
+    :returns:
+    """
+    csv_files = list()
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        try:
+            csv_files.append(func(tmp_dir))
+        except EnvironmentError:
+            pass
+
+        # Store the CSV files to the correct location
+        save_csv_files(csv_files=csv_files, using=None)
```

Comment:
I'm not happy that the way our current setup forces this duplication. I will discuss this in Amsterdam. No action on you for now.